### PR TITLE
Upgrade to apollo server 2 and client 2

### LINKED ISSUE #644: Upgrade to Apollo 2

### PR DESCRIPTION
**Pull request by rochaeb**
closes #644

### CODE REVIEW

File: packages/coinstac-api-server/src/auth-helpers.js [hunk 1]
```diff
@@ -162,24 +159,21 @@ const helperFunctions = {
...
-        })
-        .then((isUniqueEmail) => {
-          if (isUniqueEmail) {
-            res(req.payload);
-          } else if (isUniqueEmail === false) {
-            res(Boom.badRequest('Email taken'));
-          }
-        }));
+    if (isUniqueUsername) {
+      const isUniqueEmail = helperFunctions.validateUniqueEmail(req, connection);
```
Comment (kalitopix) [inline packages/coinstac-api-server/src/auth-helpers.js:167]: `const isUniqueEmail = await helperFunctions.validateUniqueEmail(req, connection);`

File: packages/coinstac-api-server/src/auth-helpers.js [hunk 2]
```diff
@@ -131,17 +131,14 @@ const helperFunctions = {
...
-        if (user.id) {
-          callback(null, true, user);
-        } else {
-          callback(null, false, null);
-        }
-      });
+  async validateToken(decoded, request) {
+    const user = await helperFunctions.getUserDetails({ username: decoded.username });
+
+    return {
```
Comment (kalitopix) [inline packages/coinstac-api-server/src/auth-helpers.js:138]: I think we'd better to use try catch for await

**Comment by kalitopix** [comment]
I think we can use `arrow functions` in react components to avoid binding methods to classes using `bind` function

**Comment by rochaeb** [comment]
This PR is outdated. It was closed in favor of #1075 .

**Pull request closed without merging**